service/tablet_allocator.cc

```diff
@@ -1049,10 +1083,40 @@ class load_balancer {
                     return r2_hosts.contains(dst.host);
                 };
 
-                lblogger.debug("Replica sets of tablets being co-located: ({}: {}), ({}, {})",
-                        global_tablet_id{table, t1.tid}, r1, global_tablet_id{table, t2.tid}, r2);
+                bool rack_awareness = rack_aware_colocation();
 
-                maybe_generate_colocation_plan({t1, std::move(r1)}, {t2, std::move(r2)}, check_constraints);
+                lblogger.debug("Replica sets of tablets being co-located: ({}: {}), ({}, {}), rack_awareness={}",
+                        global_tablet_id{table, t1.tid}, r1, global_tablet_id{table, t2.tid}, r2, rack_awareness);
+
+                if (rack_awareness) {
+                    auto r1_sets = per_rack_replica_sets(r1);
+                    auto r2_sets = per_rack_replica_sets(r2);
```

Comment:
As I commented on the previous patch, `per_rack_replica_sets` expects sorted_dc_replicas, but it doesn't seem like `get_replicas` filters the replicas by dc.

Reply:
It does, explained in your previous comment.